raiden/api/python.py

```diff
@@ -768,9 +768,9 @@ def transfer_async(
                 raise InvalidSecretOrSecretHash('secret_hash is not an hexadecimal string.')
             secret_hash = to_bytes(hexstr=secret_hash)
 
-        if secret is None and secret_hash is not None:
-            raise InvalidSecretOrSecretHash('secret_hash without a secret is not supported yet.')
-
+        # if both secret and secret_hash provided we check that sha3(secret)
+        # is giving the secert_hash. Note that it is valid to provide a secert_hash
+        # without providing a secret
```

Comment:
Note: This will conflict with https://github.com/raiden-network/raiden/pull/3797 . Our code follows the sandwich encoding as a good practice [ref. contributing](https://github.com/raiden-network/raiden/blob/master/CONTRIBUTING.md#encoding), please keep this in mind for future changes.

Reply:
can you wait with #3797 until this is merged?
also, does #3797 change the input/output of ` POST http://localhost:5001/api/v1/payments`?

Reply:
> also, does #3797 change the input/output of POST http://localhost:5001/api/v1/payments?

It shouldn't, it's just changing where we encode/decode the input/output. The values are still hex encoded with `0x` prefix.